test/test_upgrade_database.cpp

```diff
@@ -1707,6 +1706,152 @@ TEST(Upgrade_FixColumnKeys)
+        tr->commit();
+    }
+    CHECK(!File::exists(prefix + "v200.backup.realm"));
+
+    // move forward to version 202, bypassing the outlawed 201
+    _impl::GroupFriend::fake_target_file_format(std::optional<int>{202});
+    realm::BackupHandler::fake_versions({202, 200}, {{201, 0}});
+    {
+        auto hist = make_in_realm_history(path);
+        auto db = DB::create(*hist);
+        auto tr = db->start_write();
+        auto table = tr->get_table("MyTable");
+        CHECK(table->size() == 1);
+        tr->commit();
+    }
+    CHECK(File::exists(prefix + "v200.backup.realm"));
+
+    // Cleanup file and disable mockup versioning
+    File::try_remove(prefix + "v200.backup.realm");
+    _impl::GroupFriend::fake_target_file_format(std::optional<int>());
+    realm::BackupHandler::unfake_versions();
+}
+
+NONCONCURRENT_TEST(Upgrade_BackupAtoBbypassAtoC)
+{
+    SHARED_GROUP_TEST_PATH(path);
+    std::string prefix = realm::BackupHandler::get_prefix_from_path(path);
+    // clear out any leftovers from potential earlier crash of unittest
+    File::try_remove(prefix + "v200.backup.realm");
+    File::try_remove(prefix + "v201.backup.realm");
+
+    // Build a realm file with format 200
+    _impl::GroupFriend::fake_target_file_format(std::optional<int>{200});
+    realm::BackupHandler::fake_versions({200}, {});
+    {
+        auto hist = make_in_realm_history(path);
+        auto db = DB::create(*hist);
+        auto tr = db->start_write();
+        auto table = tr->add_table("MyTable");
+        table->add_column(type_String, "names");
+        tr->commit();
+    }
+
+    // upgrade to format 201
+    _impl::GroupFriend::fake_target_file_format(std::optional<int>{201});
+    realm::BackupHandler::fake_versions({201, 200}, {});
+    {
+        auto hist = make_in_realm_history(path);
+        auto db = DB::create(*hist);
+        auto tr = db->start_write();
+        auto table = tr->get_table("MyTable");
+        auto col = table->get_column_key("names");
+        table->create_object().set(col, "hr hansen");
+        tr->commit();
+    }
+    CHECK(File::exists(prefix + "v200.backup.realm"));
+
+    // upgrade further to 202, based on 201, to create a v201 backup
+    _impl::GroupFriend::fake_target_file_format(std::optional<int>{202});
+    realm::BackupHandler::fake_versions({202, 201, 200}, {});
+    {
+        auto hist = make_in_realm_history(path);
+        auto db = DB::create(*hist);
+        auto tr = db->start_write();
+    }
+    CHECK(File::exists(prefix + "v200.backup.realm"));
+    CHECK(File::exists(prefix + "v201.backup.realm"));
+
+    // downgrade/restore backup of format 200, but don't actually run the
+    // downgrade as a separate stage, but directly move forward to version 203,
+    // bypassing the outlawed 201 and 202.
+    _impl::GroupFriend::fake_target_file_format(std::optional<int>{203});
+    realm::BackupHandler::fake_versions({203, 200}, {{201, 10}});
+    {
```

Comment:
Perhaps you should comment that the "10" will prevent the backup from being deleted.